### PR TITLE
PXP-597: [CLI] Logs panel: Support search / filter for logs

### PR DESCRIPTION
**Pull request by jk-gan**
## Summary

Ticket: [PXP-597](https://mindvalley.atlassian.net/browse/PXP-597)

### Added
- [x] Add search function for logs, can be toggled by `ctrl+s`, use `Esc` to exit search
- [x] Add filter function for logs, can be toggled by `ctrl+f`, use arrow key to switch the input field, use `Esc` to exit filter

### Fixed
- [x] Fix an issue to cause repeated API calls at the first render
- [x] Fix the weird loading issue
- [x] Fix version selection issue 

[PXP-597]: https://mindvalley.atlassian.net/browse/PXP-597?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/151#" title="Author self-approved">jk-gan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by sauron-droid** [comment]
PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: cli/src/commands/tui/ui/logs.rs
Comment (mfauzaan) [inline cli/src/commands/tui/ui/logs.rs:72]: We can move this block to a function

File: cli/src/commands/tui/ui/logs.rs
Comment (jk-gan) [reply cli/src/commands/tui/ui/logs.rs:55]: done

File: cli/src/commands/tui/ui/logs.rs
Comment (jk-gan) [reply cli/src/commands/tui/ui/logs.rs:59]: done

**Comment by onimsha** [comment]
@jk-gan can you add the `Ctrl-s` and `Ctrl-f` into the header panel ?

**Comment by onimsha** [comment]
Also not sure if you notice this, but CPU usage of the Wukong CLI is exceptionally high when using Search.

This is before opening the Search box
image

And this is after doing some search
image

Is there any explanation about this behaviour of the CLI ?

**Comment by jk-gan** [comment]
qw

> Also not sure if you notice this, but CPU usage of the Wukong CLI is exceptionally high when using Search.
> 
> This is before opening the Search box image
> 
> And this is after doing some search image
> 
> Is there any explanation about this behaviour of the CLI ?

Because the search is doing local regex search (same as filter)

**Comment by jk-gan** [comment]
> @jk-gan can you add the `Ctrl-s` and `Ctrl-f` into the header panel ?

Added

**Review by onimsha — APPROVED** [review]

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/151#" title="Author self-approved">jk-gan</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/151#pullrequestreview-1654516585" title="Approved">onimsha</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
